sdk-extensions/autoconfigure-spi/build.gradle.kts

```diff
@@ -8,4 +8,5 @@ otelJava.moduleName.set("io.opentelemetry.sdk.autoconfigure.spi")
 
 dependencies {
   api(project(":sdk:all"))
+  compileOnly(project(":api:incubator"))
```

Comment:
I assume we don't see this as an issue (I don't, but calling eyes)

Reply:
It's already be a transitive dependency since this has a dependency on the SDK artifacts, which have a dependency on `:api:incubator` in order to implement experimental behavior.

Upgrading from `compileOnly` to `implementation` since `compileOnly` implies the user needs to bring their own copy which is not the case here.